torch/_dynamo/variables/builtin.py

```diff
@@ -1490,6 +1490,10 @@ def _unimplemented():
 
     # and_ is a constant fold function, so we only get here if constant fold is not valid
     def call_and_(self, tx, a, b):
+        if isinstance(a, ConstantVariable) and isinstance(b, ConstantVariable):
```

Comment:
There is actually something fishy going on. Why is constant_handler not being called? Should we not use constant_handler before trying to evaluate handler? I find this strange.

Reply:
I've changed the behaviour to fallback on constant handler. Although I still find it odd. Why is constant_handler the fallback? Why don't we try the constant handler first?? :thinking:

Reply:
I'm gonna do a refactor where we get rid of all the constant variable special-casing in individual handlers due to this poor ordering. Heads up @jansel @ezyang @lezcano